Flakes in the ClusterResourceSet unit tests

**What steps did you take and what happened:**
While investigating some unit test failures I some this error happen sometimes:

```

------------------------------
• Failure [10.167 seconds]
ClusterResourceSet Reconciler
/home/prow/go/src/sigs.k8s.io/cluster-api/exp/addons/controllers/clusterresourceset_controller_test.go:41
  Should reconcile a ClusterResourceSet when a resource is created that is part of ClusterResourceSet resources [It]
  /home/prow/go/src/sigs.k8s.io/cluster-api/exp/addons/controllers/clusterresourceset_controller_test.go:240
  Timed out after 10.001s.
      
  Expected
      <bool>: false
  to be true
  /home/prow/go/src/sigs.k8s.io/cluster-api/exp/addons/controllers/clusterresourceset_controller_test.go:308
------------------------------
....
Summarizing 1 Failure:
[Fail] ClusterResourceSet Reconciler [It] Should reconcile a ClusterResourceSet when a resource is created that is part of ClusterResourceSet resources 
/home/prow/go/src/sigs.k8s.io/cluster-api/exp/addons/controllers/clusterresourceset_controller_test.go:308
Ran 5 of 5 Specs in 29.157 seconds
FAIL! -- 4 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestAPIs (29.16s)
FAIL
FAIL	sigs.k8s.io/cluster-api/exp/addons/controllers	29.280s
```

**Environment:**

- Cluster-api version: Main

/kind bug
/area testing

--- a/exp/addons/controllers/clusterresourceset_controller_test.go
+++ b/exp/addons/controllers/clusterresourceset_controller_test.go
@@ -34,7 +34,7 @@ import (
 )
 
 const (
-	timeout              = time.Second * 10
+	timeout              = time.Second * 20
 	defaultNamespaceName = "default"
 )
 

